### PR TITLE
Replaces the IPC APC grabbing thing with a power cord and lets IPCs be genderless

### PR DESCRIPTION
**Pull request by Crazylemon64**
:cl:Crazylemon
rscadd: IPCs now come with a power cord implant to use to recharge from APCs.
tweak: IPCs can now be genderless.
/:cl:

The adjusted preferences screen looks like so:
ipc_prefs

This also allows us to easily make species genderless in the future.

### CODE REVIEW

**Comment by DesolateG** [comment]
Plug me in bby

File: code/modules/surgery/organs/augments_internal.dm [hunk 1]
```diff
@@ -8,17 +8,17 @@
 	var/implant_overlay
 	tough = 1 //not easyly broken by combat damage
 	sterile = 1 //not very germy
+	robotic = 2 // these are cybernetic after all
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/augments_internal.dm]: What does this do, precisely?

**Comment by Spacemanspark** [comment]
Where is the implant located?
How does this change how IPCs drain power?

**Comment by LightFire53** [comment]
I can now shove a cord into something and suck it dry?

File: code/modules/surgery/organs/augments_internal.dm (hunk 1 above)
Comment (Aurorablade) [inline code/modules/surgery/organs/augments_internal.dm]: Robotic decalares the organ well..robotic..rather then organic... i think theres a few levels 1 simple being kinda 'assisted', 2 being full on robotic. We could make it a define but that might be byond PR scope.

**Comment by Aurorablade** [comment]
@Spacemanspark arm implant

**Comment by Spacemanspark** [comment]
So if I cut off an IPCs arm it's gonna be fucked if I get rid of said arm and they have to go to robotics for a new one?

**Comment by fludd12** [comment]
I suggest a chest implant; It is, after all, where the microbattery is, and it makes it harder to lose it.

File: code/modules/surgery/organs/augments_internal.dm (hunk 1 above)
Comment (Crazylemon64) [inline code/modules/surgery/organs/augments_internal.dm]: It makes it possible to do robotic repair on a prosthetic if it has damage - which normally never happens outside of rare cases

**Comment by Crazylemon64** [comment]
It *is* a chest implant - it doesn't break your arm if EMP'd, it just needs to be repaired like all of your other internal components if you want to use it again.

To drain power, you use the action button in the top left corner to deploy a power cable, click on an APC, which will initiate a charging cycle if the APC has power and you still have power to charge. Then, you need to sit still while you siphon charge from the APC until your battery is full, at which point you disconnect. You can then retract the cable by clicking the action button in the top left again.

**Comment by Spacemanspark** [comment]
Hrm.
Alright.
BUT...
Add the ability to give power to an APC with this.
And make it so an IPC has issues when they run out of power. Not like a cyborg, but perhaps an even bigger slowdown than a human gets, and perhaps a random stun here and there.

**Comment by coldud13** [comment]
I hope we can lash people with the charging cable.

**Comment by pinatacolada** [comment]
...i can't get over the idea of ipc's draining or feeding other ipc's, like with a jumper cable
:5absorbing spark in the morgue

then again i don't play ipc so yeah

**Comment by taukausanake** [comment]
Yessssss. Two IPCs plugging into each other. Nothing is safe from the lewds.
As an actual feature IPCs should be able to charge each other. However I have to wonder if there will be an infinite power loop if two IPCs link to each other... Would be a cool exploit nonetheless

**Comment by DesolateG** [comment]
@taukausanake

Well.. I mean.. there are male and female ipcs..
http://www.renovation-headquarters.com/images20/male-female-plug.jpg

**Comment by taukausanake** [comment]
@DesolateG Yes, but now they have the HARDWARE

**Comment by Crazylemon64** [comment]
...

**Comment by Spacemanspark** [comment]
Wait what

**Comment by taukausanake** [comment]
Did I just kill this PR?

**Comment by Crazylemon64** [comment]
There, no more gender

**Comment by taukausanake** [comment]
You know, I don't mind. They really shouldn't have gender anyways when you think of it. And they will RP a gender either way so what does it matter?
Then again... I'm sure IPC players will still dislike. How does this show in Character Creation?

**Comment by DesolateG** [comment]
Now you're just trolling

**Comment by Aurorablade** [comment]
*smacks @taukausanake with newspaper*

**Comment by Crazylemon64** [comment]
It looks like the same as other species, except the "gender" field is omitted.
ipc_prefs

**Comment by taukausanake** [comment]
Just wanted to make sure. I can't code dive all the time even in diffs. Also the image helps

**Comment by DesolateG** [comment]
Right well, onto a serious comment. I don't support this PR.

**Comment by WhiteFlicker** [comment]
This is frankly absolutely pointless in my personal opinion. Not only because it further nerfs an already weakened race with a silly cord that, if anything, will be used for innuendo and will be a general annoyance when it inevitably ends up broken and the robotics team is dead/missing as usual. Making it a liability when finger probes function perfectly. Leaving the IPC with a limited/incapaility to recharge. AND get stunned when it can't recharge? After noting the fact that IPC get hungry considerably FASTER than any other race? No.  Let's, not. Go remove the vox's ability to float in space at will. or something.  Onto the gender subject. This is also dumb - because not only is it a completely irrelevant change, but its also silly because IPC can identify themselves as either or the other gender through personality and design. They can be gender fluid WITHOUT removing the gender system, as some switch gender at will, some other being a fixed gender because their personality/player designed them to be so.  This PR is butts all around.

**Comment by Dumbdumn5** [comment]
Don't quite care about the power cord, that's cool and all, but uh, what's with the gender bit there? Why change it? Doesn't seem to me like there's much of a point to it and that it actually removes a bit more personalization from the game that doesn't need to be removed.

**Comment by fludd12** [comment]
I'm a big fan of the power cord bit, and while I personally agree IPCs shouldn't have gender, removing it for the sake of reducing the innuendo involved with a male/female power cable is unnecessary at best. Perhaps in a different PR?

**Comment by sothangel** [comment]
The cord is interesting, but instead of removing gender altogether, why not just add a third neutral option?

**Comment by Aurorablade** [comment]
Guys.
I get the feeling the innuendo brought up earlier was making lemons uncomfortable. 
Considering the race in general this only affects predefiend emotes, this does not limit how an IPC player chooses to refer to themselves IC. You are all big creative roleplayers, you can handle that.

Let the coder do this how they they wish, Dumbdumb is a good example of some respectful feedback that lemons can take in.

**Comment by Crazylemon64** [comment]
Alright, IPCs are now able to be gendered again

**Comment by DesolateG** [comment]
How long will an ipc have to stand there to charge?

**Comment by Crazylemon64** [comment]
@WhiteFlicker What are you on about? This changes nothing about how IPCs behave when low on charge, nor is there any special code that causes IPCs to get hungrier. If you're EMP'd, you'll likely need to see a roboticist to get back on your feet, and if you're for some reason unable to get your charging system repaired, there are a couple charging stations available to use in lieu of an APC (bar bathrooms, robotics)

**Comment by Crazylemon64** [comment]
@DesolateG An IPC must stand next to the APC for 1 second for every 50 points of nutrition - or 8 seconds for 0 to 400
Normally, you'll be at around ~200-250 when you're looking for an APC to charge from, so this is more around 3-4 seconds

**Comment by Lady-Luck** [comment]
Well if you think about it, IPCs being able to be male, female, or neither makes sense - they operate on trinary.

**Comment by taukausanake** [comment]
Boo @Lady-Luck boooo

**Comment by Spacemanspark** [comment]
Spark doesn't have a gender.
Sue me

**Comment by SigholtStarsong** [comment]
This seems like a nerf to a race that, at this time, is already fairly fragile and extremely easy to kill.

**Comment by fludd12** [comment]
@SigholtStarsong Taking click delay into account, it's not much of a nerf, and gives IPCs another way to show off differences.

**Comment by Fox-McCloud** [comment]
Less snowflaked, more thematic, more OOP.

👍

**Pull request merged**